app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchTEActivity.java

```diff
@@ -361,15 +282,12 @@ protected void onDestroy() {
         super.onDestroy();
     }
 
-    @Override
-    public void onLowMemory() {
-        super.onLowMemory();
-        teiMapManager.onLowMemory();
-    }
-
     @Override
     public void onBackPressed() {
-        if (!ExtensionsKt.isKeyboardOpened(this)) {
+        if (switchOpenClose != 2) {
```

Comment:
I would send the action to the view model and this logic there in order to facilitate the testing and make view model the handler of the events.

Reply:
done!